src/mono/mono/mini/interp/transform-simd.c

```diff
@@ -563,6 +582,7 @@ emit_sri_vector128 (TransformData *td, MonoMethod *cmethod, MonoMethodSignature
 			else if (arg_size == 8) simd_intrins = INTERP_SIMD_INTRINSIC_V128_I8_CREATE_SCALAR;
 			break;
 		case SN_Equals:
+			atype = resolve_native_size (atype);
```

Comment:
why in these individual cases instead of up top?

Reply:
If you look farther up in the function there are cases where the type is checked against the parameter type.  I just made a choice to be explicit about doing where it was uses rather than introducing a new variable there.